chunks/chunks.go

```diff
@@ -57,8 +57,9 @@ type Meta struct {
 }
 
 // writeHash writes the chunk encoding and raw data into the provided hash.
-func (cm *Meta) writeHash(h hash.Hash) error {
-	if _, err := h.Write([]byte{byte(cm.Chunk.Encoding())}); err != nil {
+func (cm *Meta) writeHash(h hash.Hash, buf []byte) error {
+	buf = append(buf[:0], byte(cm.Chunk.Encoding()))
```

Comment:
Can we be consistent here and do `buf[0] = byte(...)` as we do here: https://github.com/prometheus/tsdb/pull/653/files#diff-ba6ef14901e90aea742f3fd2909d7f07R315

Reply:
As `writeHash` is not a part of `chunks.Writer`, I would not make any size assumptions of the passed buffer. It would be safer to append. The one you pointed is a `chunks.Writer` method and is assured of it's size. (But if you think it is fine as it is all internal, I can make the change)